### PR TITLE
Allow client to change boot slot via API

### PR DESCRIPTION
**Pull request by mdegat01**
## Proposed change

Every time OS is updated we keep the old version in a separate boot slot. We use this for automatic rollbacks in case of catastrophic failure but there's no reason users couldn't use this to do fast and easy manual rollbacks in cases of smaller errors they noticed which aren't catastrophic. This new API allows users to change the active boot slot between A and B

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request: https://github.com/home-assistant/developers.home-assistant/pull/2107
- Link to cli pull request: https://github.com/home-assistant/cli/pull/469

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Ruff (`ruff format supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/api/const.py [hunk 1]
```diff
@@ -60,3 +65,10 @@
 ATTR_USE_NTP = "use_ntp"
 ATTR_USERS = "users"
 ATTR_VENDOR = "vendor"
+
+
+class BootName(StrEnum):
```
Comment (agners) [inline supervisor/api/const.py:70]: I think I'd go for BootSlot instead but either way works for me

```suggestion
class BootSlot(StrEnum):
```

**Review by agners — COMMENTED** [review]
Cool, nice feature! :+1:

File: supervisor/api/const.py (hunk 1 above)
Comment (mdegat01) [reply supervisor/api/const.py:70]: @agners I was trying to stick with the names I was seeing in Rauc I guess. Rauc seemed to be ID'ing boot slots with names like `kernel.0` and `kernel.1` and referred to `A` and `B` as boot names in the status.

Comment (pvizeli) [reply supervisor/api/const.py:70]: Technical it's called boot slot over all implementation. Embedded Developers vs. Application Developers - I guess on Rauc, there where more application developers.

Comment (agners) [reply supervisor/api/const.py:70]: Reading documentation such as https://rauc.readthedocs.io/en/latest/using.html#manually-switch-to-a-different-slot seems to me that RAUC is using slot for the whole active boot "configuration" as well. Also chapter 2 is mostly talking about slots https://rauc.readthedocs.io/en/latest/basic.html 🤔

**Review by agners — APPROVED** [review]

**Comment by mdegat01** [comment]
@agners I realized the API still asked for `boot_name`. I renamed that to `boot_slot` since that's what we went with elsewhere. Let me know if you agree.

**Pull request merged**